src/compile/output/mod.rs

```diff
@@ -516,10 +525,22 @@ impl AggregatedCompilerOutput {
     }
 
     /// Whether the output contains a compiler warning
-    pub fn has_warning(&self, ignored_error_codes: &[u64]) -> bool {
+    pub fn has_warning(&self, ignored_error_codes: &[u64], ignored_file_paths: &[PathBuf]) -> bool {
         self.errors.iter().any(|err| {
             if err.severity.is_warning() {
-                err.error_code.as_ref().map_or(false, |code| !ignored_error_codes.contains(code))
+                let is_code_not_ignored = err
+                    .error_code
+                    .as_ref()
+                    .map_or(false, |code| !ignored_error_codes.contains(code));
+
+                let is_file_not_ignored = err.source_location.as_ref().map_or(false, |location| {
+                    !ignored_file_paths.contains(&PathBuf::from(&location.file))
```

Comment:
this should not be `contains`, instead we want to check if the file is inside one fo the ignored files (`starts_with`)